src/socket.rs

```diff
@@ -1294,6 +1318,41 @@ mod test {
     }
 
     #[test]
+    fn test_bind_and_connect_with_udp_socket() {
+        use std::net::{UdpSocket, Ipv4Addr, SocketAddrV4};
+
+        let server_udp_socket = iotry!(UdpSocket::bind("0.0.0.0:0"));
+        let client_udp_socket = iotry!(UdpSocket::bind("0.0.0.0:0"));
+
+        let server_port = iotry!(server_udp_socket.local_addr()).port();
+        let server_addr = SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), server_port);
+
+        let mut server = iotry!(UtpSocket::bind_with_udp_socket(server_udp_socket));
+        assert!(server.state == SocketState::New);
+
+        thread::spawn(move || {
+            let mut client = iotry!(UtpSocket::connect_with_udp_socket(client_udp_socket, server_addr));
+            assert!(client.state == SocketState::Connected);
+            // Check proper difference in client's send connection id and receive connection id
+            assert_eq!(client.sender_connection_id, client.receiver_connection_id + 1);
+            assert_eq!(client.connected_to,
+                       server_addr.to_socket_addrs().unwrap().next().unwrap());
+            iotry!(client.close());
+            drop(client);
+        });
+
+        let mut buf = [0u8; BUF_SIZE];
+        match server.recv_from(&mut buf) {
```

Comment:
what's the purpose of this `match`? Why not just:

```rust
println!("{:?}", server.recv_from(&mut buf));
```

But anyways, the `println!` doesn't test anything, so I assume you used it to "debug" something and forgot to take it away afterwards.